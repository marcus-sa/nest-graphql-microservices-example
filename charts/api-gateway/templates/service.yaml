apiVersion: v1
kind: Service
metadata:
  name: {{ include "api-gateway.fullname" . }}
  labels:
{{ include "api-gateway.labels" . | indent 4 }}
spec:
  type: {{ .Values.service.type }}
  ports:
    - port: {{ .Values.service.port }}
      targetPort: http
      protocol: TCP
      name: http
  selector:
    app.kubernetes.io/name: {{ include "api-gateway.name" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
