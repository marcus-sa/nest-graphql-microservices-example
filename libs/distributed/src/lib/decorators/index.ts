export * from './resolve-reference.decorator';