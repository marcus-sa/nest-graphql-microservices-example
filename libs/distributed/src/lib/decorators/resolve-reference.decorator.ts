import { SetMetadata } from '@nestjs/common';
import { RESOLVE_REFERENCE_METADATA } from '../tokens';

export function ResolveReference(): MethodDecorator {
	return (target: Object | Function, key?: string | symbol, descriptor?: any) => {
		SetMetadata(RESOLVE_REFERENCE_METADATA, true)(target, key, descriptor);
	};
}