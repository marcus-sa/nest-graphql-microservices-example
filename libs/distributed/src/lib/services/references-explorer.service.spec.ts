import { INestApplicationContext, Module } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';
import { Resolver } from '@nestjs/graphql';
import { ReferencesExplorerService } from './references-explorer.service';
import { ResolveReference } from '../decorators';
import { MetadataScanner } from '@nestjs/core/metadata-scanner';
import { GRAPHQL_MODULE_OPTIONS } from '@nestjs/graphql/dist/graphql.constants';

@Resolver()
class AccountsResolver {
  @Resolver('User')
  @ResolveReference()
  user() {}
}

@Module({
  providers: [
    {
      provide: GRAPHQL_MODULE_OPTIONS,
      useValue: {},
    },
    AccountsResolver,
    MetadataScanner,
    ReferencesExplorerService,
  ],
})
class AppModule {}

describe('ReferencesExplorerService', () => {
  let app: INestApplicationContext;
  let referencesExplorer: ReferencesExplorerService;

  beforeEach(async () => {
    app = await NestFactory.createApplicationContext(AppModule);
    referencesExplorer = app.get(ReferencesExplorerService);
  });

  describe('explore', () => {
    it('should return User with __resolveReference', () => {
      const resolvers = referencesExplorer.explore();

      expect(resolvers).toEqual({
        User: {
          __resolveReference: expect.any(Function),
        }
      });
      expect(resolvers).toMatchSnapshot();
    });
  });
});