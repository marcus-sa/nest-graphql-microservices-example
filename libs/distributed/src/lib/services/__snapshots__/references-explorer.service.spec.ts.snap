// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferencesExplorerService explore should return User with __resolveReference 1`] = `
Object {
  "User": Object {
    "__resolveReference": [Function],
  },
}
`;

exports[`ReferencesExplorerService explore should return resolver with __resolveReference property 1`] = `
Object {
  "User": Object {
    "__resolveReference": [Function],
  },
}
`;
